internal/bminventory/inventory.go

```diff
@@ -340,9 +341,36 @@ func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context
 		}
 	}
 
+	params.NewClusterParams.NetworkType, err = getDefaultNetworkType(params)
+	if err != nil {
+		return params, err
+	}
+
 	return params, nil
 }
 
+// If the cluster is SNO or the OpenShift version >= 4.12,
+// the network type will be set to "OVNKubernetes", otherwise it will be set to "OpenShiftSDN"
+func getDefaultNetworkType(params installer.V2RegisterClusterParams) (*string, error) {
+	// If the network type is specified already, we return it
+	if params.NewClusterParams.NetworkType != nil {
+		return params.NewClusterParams.NetworkType, nil
+	}
+
+	isOpenShiftVersionRecentEnough, err := common.VersionGreaterOrEqual(swag.StringValue(params.NewClusterParams.OpenshiftVersion), minimalOpenShiftVersionForDefaultNetworkTypeOVNKubernetes)
+	if err != nil {
+		return params.NewClusterParams.NetworkType, err
```

Comment:
I may be too picky, but I find it slightly confusing. We enter the error-path here and return effectively a tuple (nil,err) but instead of nil we write it as params.somethingsomething. The first argument is effectively nil because if it wasn’t, we would have caught it in the conditional two lines above. So if this is true, could we just write here “return nil, err”? To anyone consuming this function in the future it will be clear that this path is not a good one